File: Software Development Module/Week4/BakeryApp.java
/*
BakeryApp.java
Billy Cussen 
09/07/2020
*/

//This is the App Class for Bakery.java. This is the executable class

import java.util.InputMismatchException;
import java.util.Scanner;

public class BakeryApp {
    
    public static void main(String[]args){

        //Variables
        int items = 0;
        int boxes;
        int trips;
        boolean inputCheckItems;

        //Objects
        Bakery myBakery;
        Scanner keyboard;

        //Input
        do{

            keyboard = new Scanner(System.in);
            inputCheckItems = true;

            try{
                System.out.print("Please enter how many Items there are today: ");
                items = keyboard.nextInt();
            }catch(InputMismatchException e){
                System.out.println("You need to enter an Integer (e.g. 4). Please try again: ");
                inputCheckItems = false;
            }

        }while(!inputCheckItems);

        keyboard = null;

        //Process
        myBakery = new Bakery(items);
        myBakery.computeBoxesAndTrips();
        
        //Output
        boxes = myBakery.getBoxes();
        trips = myBakery.getTrips();
        myBakery = null;

        System.out.println("Total Items: "+items+"\nTotal Boxes: "+boxes+"\nTotal Trips: "+trips);
    }
}

File: Python Course/Software Development Module - in Python/Week4/BakeryApp.py
"""
BakeryApp.py
Billy Cussen
27/09/2020
"""
import gc
import Bakery
from Bakery import myBakery

repeat = "y"

while repeat == "y":

    inputCheck = True

    #Input - Get Items
    while inputCheck:

        try:
            items = int(input("Please enter how many items there are today: "))
        except:
            print("You need to enter an Integer (e.g. 1, 12)!")
        else:
            if items < 0:
                print ("You can't enter less than 0")
            else:
                inputCheck = False

    #Process & Output
    myBakeryApp = myBakery(items)
    myBakeryApp.calculateBoxesAndTrips()
    print("Total Boxes today: "+str(myBakeryApp.boxes))
    print("Total Trips today: "+str(myBakeryApp.trips))
    #GC to Free up Memory
    myBakeryApp = None
    gc.collect()

    repeat = ""
    #Input - Retry?
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter if you would like to retry or not (\"Y\" or \"N\"): "))
        repeat = repeat.lower()
        if repeat != "y" and repeat != "n":
            print ("You need to enter \"Y\" or \"N\"!")

print("Thanks for using my App!")
            